packages/react-reconciler/src/ReactFiberCompleteWork.new.js

```diff
@@ -862,6 +862,14 @@ function completeWork(
     }
     case HostRoot: {
       const fiberRoot = (workInProgress.stateNode: FiberRoot);
+
+      if (enableTransitionTracing) {
+        const transitions = workInProgress.memoizedState.transitions;
```

Comment:
We only want to schedule an effect if there was actually a new transition. This will schedule an effect even if nothing changed.

Reply:
Oooh good catch